incorporating-accessibility-into-your-dev-process/demo/src/ButtonsAndLinks/ButtonsAndLinks.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useRef } from "react";
+
+import styles from "./ButtonsAndLinks.module.css";
+
+const ButtonsAndLinks = () => {
+  const firstDialog = useRef<HTMLDialogElement>(null);
```

Comment:
Why do it this way vs some state value?

Is there a benefit to having the markup in the document if it isn't rendered?

Reply:
Could you expand on this? I don’t quite understand.

For this component, I wanted a ref to the `<dialog>` so I can call the native imperative `showModal()` and `close()` APIs.

Reply:
For sure. It just looks and feels super un-reacty.

Toggling a components state in react is usually tied to some react state that conditionally renders the modal in one way or another which gives a reacty interface like (example from mui):
```
export default function BasicModal() {
  const [open, setOpen] = useState(false);
  const handleOpen = () => setOpen(true);
  const handleClose = () => setOpen(false);

  return (
    <div>
      <Button onClick={handleOpen}>Open modal</Button>
      <Modal
        open={open}
        onClose={handleClose}
        aria-labelledby="modal-modal-title"
        aria-describedby="modal-modal-description"
      >
        ... modal content ...
      </Modal>
    </div>
  );
}
```
Most commonly this conditional rendering is facilitated through portal (for material: https://github.com/mui/material-ui/blob/master/packages/mui-material/src/Modal/Modal.js#L199)

Looking into it, Imperative DOM node access is the only way to use the dialog element. However, Imperative dom node access is generally something we like to avoid _(generally)_.

The reason i asked about whether there is an advantage to it being in the markup or not is because with these conditional renders you can make it so things are or aren't in the dom, but this dialog's markup is always there.

Reply:
Ah, I see what you mean. I want to use the browser’s native `<dialog>` element… in general, I want to take advantage of platform features so I don’t have to implement them myself or bring in a dependency. Do you have concerns about that?